### ISSUE TITLE
Lookup by common name is returning 404

### ISSUE THREAD

**Issue by bbengfort** [trisacrypto/testnet#16]
On the live server lookup by common name is returning 404 even though lookup by id is working and returning the common name that was requested.

**Comment by bbengfort** [comment]
The main issue was that the Lookup method was using "common_name" instead of "name", but on following up on this, there appears to be synchronization errors with both the name and countries index that need to be resolved.

**Comment by bbengfort** [comment]
The primary issue was that synchronizing the index to disk was in the shutdown method but the main goroutine was exiting before the shutdown routine could complete, meaning that the indices did not get saved to disk. I modified the `Serve` method to wait until the shutdown was complete by blocking on an error channel. 

It was also a good opportunity to enhance the indexing process, and the index has to be repaired on the production server anyway. To that end, I beefed up the index data structures and added some tests for them, then added a Reindex method and a check if the indices were empty to perform the reindex. If it's a brand new server - fine, the reindex will iterate over nothing; but if the indices were incorrectly loaded at least they will be rebuilt on start.
